core/execution/market.go

```diff
@@ -1731,7 +1732,27 @@ func (m *Market) confirmMTM(
 				m.log.Error("Failed to perform bond slashing",
 					logging.Error(err))
 			}
-			m.broker.Send(events.NewLedgerMovements(ctx, transfers))
+			// if bond slashing occurred then amounts in "closed" will not be accurate
+			if len(transfers) > 0 {
+				m.broker.Send(events.NewLedgerMovements(ctx, transfers))
+				asset, _ := m.mkt.GetAsset()
+				closedRecalculated := make([]events.Margin, 0, len(closed))
+				for _, c := range closed {
+					recalculated := c
+					if pos, ok := m.position.GetPositionByPartyID(c.Party()); ok {
+						margin, err := m.collateral.GetPartyMargin(pos, asset, m.mkt.ID)
+						if err != nil {
+							m.log.Error("couldn't get party margin",
+								logging.PartyID(c.Party()),
+								logging.Error(err))
+							continue
+						}
+						recalculated = margin
```

Comment:
why the unecessary copy here?
just happen margin in closedRecalculated instead?

Reply:
Wanted to keep the old value in case we weren't able to recalculate - so we don't make the "closed" list shorter. Replaced with two separate appends now.